### PR TITLE
fix snapsync issue with forest

### PR DESCRIPTION
**Pull request by matkt**
## PR description

Fixing an issue during snapsync with forest related to the heal step

the step checking whether the node existed or not during the healing process could lead to a null pointer exception due to the null nodeHash, which was only utilized within the context of forest.

### CODE REVIEW

**Comment by matkt** [comment]
Tests done

Scenario/StorageFormat | BONSAI | FOREST
-- | -- | --
Initial Snapsync | ✅ | ✅

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/WorldStateKeyValueStorage.java [hunk 1]
```diff
@@ -78,6 +78,11 @@ private Optional<Bytes> getTrieNode(final Bytes32 nodeHash) {
     }
   }
 
+  @Override
+  public Optional<Bytes> getUnSafeTrieNode(final Bytes key) {
+    return keyValueStorage.get(key.toArrayUnsafe()).map(Bytes::wrap);
+  }
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/WorldStateKeyValueStorage.java:84]: a style suggestion, rename method to `getTrieNodeUnsafe` to uniform with other _unsafe_ methods like `toArrayUnsafe`

**Review by fab-10 — APPROVED** [review]
LGTM this fix the sync on the Sepolia instance, just a style suggestion.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/WorldStateKeyValueStorage.java (hunk 1 above)
Comment (matkt) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/WorldStateKeyValueStorage.java:84]: done

**Pull request merged**